prometheus-openstack-exporter

```diff
@@ -170,15 +170,18 @@ class DataGatherer(Thread):
         info['aggregates'] = [x.to_dict() for x in nova.aggregates.list()]
 
         info['instances'] = []
-        marker = ''
-        while True:
-            search_opts = {'all_tenants': '1', 'limit': '100', 'marker': marker}
-            new_instances = [x._info for x in nova.servers.list(search_opts=search_opts)]
-            if new_instances:
-                marker = new_instances[-1]['id']
-                info['instances'].extend(new_instances)
-            else:
-                break
+        # Exclude instances in 'BUILD' state as they cannot be used as markers:
+        #   https://github.com/CanonicalLtd/prometheus-openstack-exporter/issues/90
+        for status in 'ACTIVE', 'SHUTOFF', 'ERROR', 'VERIFY_RESIZE', 'SHELVED_OFFLOADED', 'SUSPENDED':
```

Comment:
Any reason not to have this be an alphabetically sorted list? Seems more idiomatic.

Reply:
Fixed